### PR TITLE
CTW-370/patientid-guard

### PR DESCRIPTION
**Pull request by pmadria3**
This PR adds a guard for if the patientID doesn't exist in Zus. 

Screen Shot 2022-10-06 at 8 58 11 AM

### CODE REVIEW

File: src/components/core/alert.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+      <img src={alert} alt="alert" /> Conditions Unavailable
+    </div>
+    <div className="ctw-text-caution-lightred">
+      We are unable to access Condition information for this patient.
+    </div>
+    <div className="ctw-text-caution-lightred">
+      Contact your system administrator or customer service for assistance.
+    </div>
+  </div>
+);
```
Comment (luthfib) [inline src/components/core/alert.tsx:15]: I imagine we are going to be using this AlertDialog in other components so it should be generic and take in the message it should

File: .changeset/silent-squids-divide.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@zus-health/ctw-component-library": patch
+---
+
+Don't fetch when not on patient provider. Also fixes overfetching
```
Comment (luthfib) [inline .changeset/silent-squids-divide.md:5]: Fix changeset, this is not the correct description

**Comment by luthfib** [comment]
Nit: branch names should be projectPrefix-ProjectNumber/description-of-project

**Review by luthfib — CHANGES_REQUESTED** [review]
Looking good, but some changes needed. Need padding around the alert dialog. Refer to figma and compare it yours.

File: src/components/core/alert.tsx (hunk 1 above)
Comment (luthfib) [reply src/components/core/alert.tsx:15]: Alert Icon and the title need to be aligned. Also the paragraph should not appear right below the alert icon but to the side. Please refer to figma

**Review by luthfib — COMMENTED** [review]
I believe we dont need to add a new icon but can exclamation-triangle from heroicons.

File: src/components/content/conditions.tsx [hunk 3]
```diff
@@ -183,79 +185,83 @@ export function Conditions({ className }: ConditionsProps) {
...
+          <button
+            type="button"
+            className="ctw-btn-clear ctw-link"
+            onClick={handleAddNewCondition}
+          >
+            + Add Condition
+          </button>
+        )}
       </div>
+      {patientResponse.error && <AlertDialog />}
```
Comment (luthfib) [inline src/components/content/conditions.tsx:198]: Need a fragment around this to get rid of the error

Comment (luthfib) [inline src/components/content/conditions.tsx:198]: you can do patientResponse.isError for more succint error handling

File: src/components/content/conditions.tsx [hunk 4]
```diff
@@ -183,79 +184,92 @@ export function Conditions({ className }: ConditionsProps) {
...
-          <div className="ctw-conditions-title-container">
-            <div className="ctw-title">Patient Record</div>
-            <ToggleControl
-              onFormChange={handleFormChange}
-              toggleProps={{ name: "conditions", text: "Include Inactive" }}
+      <>
+        {patientResponse.isError && (
+          <div className="ctw-p-5">
+            <AlertDialog
+              resourceType="Condition"
```
Comment (luthfib) [inline src/components/content/conditions.tsx:201]: Doesnt this need to be plural?

**Comment by pmadria3** [comment]
Going with the design team icon after discussion

**Review by luthfib — CHANGES_REQUESTED** [review]
These still do not match. Look at the aligement with the icon. Align center should the the tricky if the icon and the title are in the same flex div. Also the content shouldnt be below the alert icon

image


image

File: src/styles/tailwind.theme.ts [hunk 5]
```diff
@@ -45,6 +45,8 @@ export const TailwindTheme = {
     caution: {
       main: "#F59E0B",
       light: "#FEF3C7",
+      burgundy: "#92400E",
```
Comment (luthfib) [inline src/styles/tailwind.theme.ts:48]: Still checking if we want to add new colors, if we do I think we could change these to be:

```
burgundy/text
lightred/body
yellow/default
```

The current names dont have too much of a semantic meaning

File: src/components/content/conditions.tsx [hunk 6]
```diff
@@ -183,79 +184,92 @@ export function Conditions({ className }: ConditionsProps) {
...
       <div className="ctw-flex ctw-h-11 ctw-items-center ctw-justify-between ctw-bg-bg-light ctw-p-3">
         <div className="ctw-title">Conditions</div>
-        <button
-          type="button"
-          className="ctw-btn-clear ctw-link"
-          onClick={handleAddNewCondition}
-        >
-          + Add Condition
-        </button>
+        {!patientResponse.isError && (
```
Comment (WesleyKapow) [inline src/components/content/conditions.tsx:187]: Might be easier to have do:

```ts
if (patientResponse.isError) {
  return (<...>)
}

return ( ...) // The existing stuff.
```

File: src/components/core/alert.tsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+export type AlertDialogProps = {
+  resourceType: string;
+  children: ReactNode;
+};
+
+export const AlertDialog = ({ resourceType, children }: AlertDialogProps) => (
+  <div className="ctw-flex ctw-items-start ctw-space-x-2 ctw-bg-caution-bg ctw-p-4">
+    <img src={alert} alt="alert" className="ctw-shrink-0" />
+    <div className="ctw-space-y-2">
+      <div className="ctw-flex ctw-text-caution-heading">
```
Comment (WesleyKapow) [inline src/components/core/alert.tsx:13]: flex is usually not needed when there's only one child.. flex box is about how elements line up in a row or column, one child = nothing to do.

File: src/components/core/alert.tsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import { ReactNode } from "react";
+import alert from "./alert.svg";
+
+export type AlertDialogProps = {
+  resourceType: string;
+  children: ReactNode;
+};
+
+export const AlertDialog = ({ resourceType, children }: AlertDialogProps) => (
```
Comment (WesleyKapow) [inline src/components/core/alert.tsx:9]: You've made this alert component specific to resource unavailable errors. We may want to show it for other things too.

File: .changeset/silent-squids-divide.md [hunk 9]
```diff
@@ -0,0 +1,5 @@
+---
+"@zus-health/ctw-component-library": patch
+---
+
+Adds an error display to the conditions table if there is no patientID corresponding in Zus Health.
```
Comment (WesleyKapow) [inline .changeset/silent-squids-divide.md:5]: Maybe change to `... conditions table if patient could not be found".

**Review by WesleyKapow — APPROVED** [review]

**Pull request merged**